File: backed/src/main/java/com/knowledge/entity/ProjectTemplate.java
package com.knowledge.entity;

import com.knowledge.common.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 项目模板实体类
 *
 * @author system
 * @date 2025-10-06
 */
@Data
@EqualsAndHashCode(callSuper = true)
public class ProjectTemplate extends BaseEntity {
    
    /**
     * 项目分类/领域
     */
    private String domain;
    
    /**
     * 模板名称
     */
    private String name;
    
    /**
     * 模板描述
     */
    private String description;
    
    /**
     * 功能性需求（JSON格式）
     */
    private String requirements;
    
    /**
     * 非功能性需求/通用特性（JSON格式）
     */
    private String commonFeatures;
}

File: fronted/src/api/projectTemplate.ts
import request from './request'

export interface ProjectRequirement {
  name: string
  desc: string
  children?: ProjectRequirement[]
}

export interface ProjectTemplate {
  id?: number
  domain: string
  name: string
  description: string
  requirements: string  // JSON string
  commonFeatures: string  // JSON string
  recCreator?: string
  recCreateTime?: string
  recRevisor?: string
  recReviseTime?: string
  archFlag?: number
}

/**
 * 获取所有项目模板
 */
export const getAllProjectTemplates = () => {
  return request.get('/projecttemplate/list')
}

/**
 * 根据ID获取项目模板
 */
export const getProjectTemplateById = (id: number) => {
  return request.get(`/projecttemplate/${id}`)
}

/**
 * 根据领域获取项目模板
 */
export const getProjectTemplatesByDomain = (domain: string) => {
  return request.get(`/projecttemplate/domain/${domain}`)
}

/**
 * 创建项目模板
 */
export const createProjectTemplate = (projectTemplate: ProjectTemplate) => {
  return request.post('/projecttemplate', projectTemplate)
}

/**
 * 更新项目模板
 */
export const updateProjectTemplate = (projectTemplate: ProjectTemplate) => {
  return request.put('/projecttemplate', projectTemplate)
}

/**
 * 删除项目模板
 */
export const deleteProjectTemplate = (id: number) => {
  return request.delete(`/projecttemplate/${id}`)
}

